bots/task/__init__.py

```diff
@@ -194,7 +194,7 @@ def begin(publish, name, context, issue):
 
     return publishing
 
-def finish(publishing, ret, name, context, issue, run_tests):
+def finish(publishing, ret, name, context, issue):
```

Comment:
This looks like a revert from a previous commit. Could you please split that out with a separate rationale, and then add the retry logic above on top of it?